device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSenderLinkHandler.java

```diff
@@ -207,14 +207,14 @@ AmqpsSendResult sendMessageAndGetDeliveryTag(MessageImpl protonMessage)
             }
 
             log.trace("Message was sent over {} sender link with delivery tag {} and hash {}", getLinkInstanceType(), new String(deliveryTag), delivery.hashCode());
-            return new AmqpsSendResult(true, deliveryTag);
+            return new AmqpsSendResult(deliveryTag);
         }
         catch (Exception e)
         {
             log.warn("Encountered a problem while sending a message on {} sender link with link correlation id {}", getLinkInstanceType(), this.linkCorrelationId, e);
             this.senderLink.advance();
             delivery.free();
-            return new AmqpsSendResult(false);
```

Comment:
if the passed in value was `false` then why in the AmqpsSendResult constructor we remove the boolean and set it to true? I am a bit confused.

Reply:
AmqpsSendResult has 2 constructors 
-> no params = delivery unsuccessful (with a default delivery tag), 
-> delivery tag as param = delivery successful.